erpnext/stock/doctype/stock_reconciliation/stock_reconciliation.py

```diff
@@ -42,23 +50,29 @@ def remove_items_with_no_change(self):
 		"""Remove items if qty or rate is not changed"""
 		self.difference_amount = 0.0
 		def _changed(item):
-			qty, rate = get_stock_balance(item.item_code, item.warehouse,
-					self.posting_date, self.posting_time, with_valuation_rate=True)
-			if (item.qty==None or item.qty==qty) and (item.valuation_rate==None or item.valuation_rate==rate):
+			item_dict = get_stock_balance_for(item.item_code, item.warehouse,
+				self.posting_date, self.posting_time, batch_no=item.batch_no)
+
+			if ((item.qty==None or item.qty==item_dict.get("qty"))
+				and (item.valuation_rate==None or item.valuation_rate==item_dict.get("rate"))
```

Comment:
`not item.valuation_rate`??

Reply:
item.valuation_rate can be zero and not item.valuation_rate will be true in that case

Reply:
item.qty is None

Reply:
Done